File: trunk/sketchbook/serialport_test.rb
#!/usr/bin/env ruby

require "serial_port"

#@sp = Funnel::SerialPort.new('COM3', 38400)
@sp = Funnel::SerialPort.new('/dev/cu.usbserial-A30009cb', 38400)

def talk(command, length)
  @sp.write(command)
  while @sp.bytes_available < length do
    sleep(0.001)
  end
  @sp.read(length)
end

p talk('Q*', 2)
sleep(0.1)
p talk('?*', 10)
p talk('KONFIGURATION_1*', 16)
sleep(0.1)

puts "*** 2 bytes version ***"
start = Time.now
1000.times do
  talk('h*', 2)
  talk('l*', 2)
end
finish = Time.now
puts "#{1000 / (finish - start)} fps"

puts ""

sleep(1)

def padding(s, packetsize)
  s + ("\000" * (packetsize - s.size))
end

on = padding('h*', 62)
off = padding('l*', 62)

puts "*** #{on.size} bytes version ***"
start = Time.now
1000.times do
  talk(on, 2)
  talk(off, 2)
end
finish = Time.now
puts "#{1000 / (finish - start)} fps"

# 38400bps = 3840cps
# 64 / 3840 = 16ms
# 1000 / (16 + 16) = 32 fps

# 115000bps = 11500cps
# 64 / 11500 = 6ms
# 1000 / (6 + 6) = 83 fps

File: serialport_test.py
#!/usr/bin/env python
# encoding: utf-8
"""
untitled.py
"""

import serial
import time
import timeit

sp = serial.Serial('/dev/cu.usbserial-A30009cb', 38400, timeout = 1)
print sp.portstr
print sp

sp.write('Q*')
time.sleep(0.1)
print sp.read(2)

sp.write('?*')
print sp.read(10)

sp.write('KONFIGURATION_1*')
print sp.read(16)
time.sleep(0.1)

def blink():
	sp.write('h*')
	sp.read(2)
	sp.write('l*')
	sp.read(2)

start = time.time()

#t = timeit.Timer('blink()')
#t.timeit(1000)

for i in range(1000):
	blink()

finish = time.time()
print finish - start

def main():
	pass

if __name__ == '__main__':
	main()
